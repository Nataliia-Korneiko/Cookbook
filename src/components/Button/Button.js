/* eslint-disable react/button-has-type */
import React from 'react';
import PropTypes from 'prop-types';
import s from './Button.module.css';

const Button = ({ type = 'button', label = '', onClick = () => null }) => {
  return (
    <button className={s.button} type={type} onClick={onClick}>
      {label}
    </button>
  );
};

Button.propTypes = {
  type: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default Button;
