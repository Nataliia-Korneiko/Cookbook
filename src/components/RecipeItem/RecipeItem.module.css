.recipe {
  padding: 10px 15px 15px;
  border-top: 5px solid transparent;
  border-radius: 5px;
  width: 300px;
  background-color: var(--color-white);
  background-image: url(../../assets/img/backdrops/backdrop-06.png);
  background-repeat: no-repeat;
  background-position: center;
}

.low__level {
  border-color: var(--color-green);
}

.normal__level {
  border-color: var(--color-blue);
}

.high__level {
  border-color: var(--color-red);
}

.spans {
  min-height: 38px;
  font-size: 14px;
}

.span {
  display: block;
}

.create__date {
  color: var(--color-dark-green);
}

.edit__date {
  color: var(--color-dark-red);
}

.wrapper {
  margin-bottom: 15px;
  padding: 10px;
  display: flex;
  flex-direction: column;
  height: 300px;
  background-color: var(--bg-color-white);
  box-shadow: var(--box-shadow-light);
}

.title {
  font-family: 'Marck Script', cursive;
  font-size: 22px;
  font-weight: 800;
  color: var(--color-golden);
  text-align: center;
}

.title::after {
  margin: 0 auto;
  display: block;
  content: '';
  width: 247px;
  height: 24px;
  background-image: url(../../assets/img/decoration/decoration.png);
  background-repeat: no-repeat;
  background-position: center;
}

.description {
  text-align: justify;
  overflow-y: scroll;
}

.buttons {
  display: flex;
  justify-content: space-between;
}

@media (max-height: 640px) {
  .wrapper {
    height: 192px;
  }
}
