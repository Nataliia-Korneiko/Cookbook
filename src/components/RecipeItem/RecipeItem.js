import React from 'react';
import PropTypes from 'prop-types';
import Button from '../ButtonItem/ButtonItem';
import s from './RecipeItem.module.css';

const RecipeItem = ({
  text,
  description,
  level,
  createDate,
  editDate,
  onDeleteRecipe,
  onEditRecipe,
}) => (
  <div className={`${s.recipe} ${s[`${level}__level`]}`}>
    <div className={s.spans}>
      <span className={`${s.span} ${s.create__date}`}>{createDate}</span>
      <span className={`${s.span} ${s.edit__date}`}>{editDate}</span>
    </div>
    <div className={s.wrapper}>
      <h2 className={s.title}>{text}</h2>
      <p className={s.description}>{description}</p>
    </div>
    <div className={s.buttons}>
      <Button label="Edit" type="button" onClick={onEditRecipe} />
      <Button label="Delete" type="button" onClick={onDeleteRecipe} />
    </div>
  </div>
);

RecipeItem.defaultProps = {
  createDate: '',
  editDate: '',
};

RecipeItem.propTypes = {
  text: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  level: PropTypes.string.isRequired,
  createDate: PropTypes.string,
  editDate: PropTypes.string,
  onDeleteRecipe: PropTypes.func.isRequired,
  onEditRecipe: PropTypes.func.isRequired,
};

export default RecipeItem;
