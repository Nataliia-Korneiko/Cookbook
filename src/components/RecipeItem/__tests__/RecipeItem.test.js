/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { shallow } from 'enzyme';
import RecipeItem from '../RecipeItem';
import Button from '../../ButtonItem/ButtonItem';

describe('RecipeItem component', () => {
  const props = {
    text: '',
    description: '',
    level: '',
    createDate: '',
    editDate: '',
    onDeleteRecipe: () => null,
    onEditRecipe: () => null,
  };

  const newRecipeItem = shallow(<RecipeItem {...props} />);

  it('renders a properly', () => {
    expect(newRecipeItem).toMatchSnapshot();
  });

  it('renders four div', () => {
    expect(newRecipeItem.find('div')).toHaveLength(4);
  });

  it('renders two span', () => {
    expect(newRecipeItem.find('span')).toHaveLength(2);
  });

  it('renders one h2', () => {
    expect(newRecipeItem.find('h2')).toHaveLength(1);
  });

  it('renders one p', () => {
    expect(newRecipeItem.find('p')).toHaveLength(1);
  });

  it('renders two Button', () => {
    expect(newRecipeItem.find(Button)).toHaveLength(2);
  });
});
