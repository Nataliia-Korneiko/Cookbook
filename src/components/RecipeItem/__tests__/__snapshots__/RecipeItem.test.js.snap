// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeItem component renders a properly 1`] = `ShallowWrapper {}`;
